lib/plugin/compilers/i18n.json.coffee

```diff
@@ -4,124 +4,147 @@ helpers = share.helpers
 compilers = share.compilers
 compiler_configuration = share.compiler_configuration
 
-compilers.i18n_json = (compileStep) ->
-  compiler_configuration.registerInputFile(compileStep)
```

Comment:
@Slava I think failing test are mostly related to this kind of construct. Here the package keeps a registered list of files it has seen but I haven’t re-implemented this with the caching compiler. I guess I could use `processFilesForTarget` for that purpose (you still get the whole list even if only one package change, right?

Also this shared state in memory was used to share configuration between different compilers (like the one handling the all `*.i18.json` files and the one handling the `package-tap.i18n` configuration file), is there still a way to share some state between these two separate compilers, or do I need to merge them into one and dispatch the actual transforms after checking the file name in `compileOneFile` (to distinguish between translation file and configuration files)?